src/pipeline_edsl/pipeline_library.ml

```diff
@@ -22,6 +22,114 @@ module Input = struct
   let of_bam ?fragment_id ?sorted ~reference_build how s =
     (fragment_id, Of_bam (how, sorted, reference_build,  s))
   let fastq_sample ~sample_name files = Fastq {sample_name; files}
+
+  let tag_v0 = "Biokepi_input_v0"
+  let current_version_tag = tag_v0
+
+  let to_yojson =
+    let string s = `String s in
+    let option f o : Yojson.Safe.json = Option.value_map o ~default:`Null ~f in
+    let data_to_yojson =
+      function
+      | PE (r1, r2) ->
```

Comment:
"SE", "PE" are very common abbreviations in the field.

I changed the JSON names though, to expand the acronym there, but not the API:

```````````````json
{
  "biokepi-input-v0": {
    "fastq": {
      "sample-name": "Sample2",
      "fragments": [
        {
          "fragment-id": "frg1",
          "data": {
            "paired-end": { "r1": "frg1R1.fastq", "r2": "frg1R2.fastq" }
          }
        },
        { "fragment-id": "frg2", "data": { "single-end": "frg2.fastq" } },
        {
          "fragment-id": null,
          "data": {
            "bam": {
              "kind": "paired-end",
              "sorting": "coordinate",
              "reference-genome": "b37",
              "path": "n2.bam"
            }
          }
        }
      ]
    }
  }
}
```````````````